### PR TITLE
feat: Borrow

### PR DESCRIPTION
**Pull request by SteveYuOWO**

### CODE REVIEW

File: src/AAVE/Modal/RepayModal.jsx [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+                                : "$ -"}
+                            </WhiteTexture>
+                          </div>
+                        ),
+                      }}
+                    />
+                    <Widget
+                      src={`${config.ownerId}/widget/AAVE.Modal.FlexBetween`}
+                      props={{
+                        left: <PurpleTexture>Health factor</PurpleTexture>,
```
Comment (think-in-universe) [inline src/AAVE/Modal/RepayModal.jsx:471]:
```suggestion
                        left: <PurpleTexture>Health Factor</PurpleTexture>,
```

File: src/AAVE/Modal/BorrowModal.jsx [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+              src={`${config.ownerId}/widget/AAVE.Modal.RoundedCard`}
+              props={{
+                title: "Transaction Overview",
+                config,
+                children: (
+                  <TransactionOverviewContainer>
+                    <Widget
+                      src={`${config.ownerId}/widget/AAVE.Modal.FlexBetween`}
+                      props={{
+                        left: <PurpleTexture>Health factor</PurpleTexture>,
```
Comment (think-in-universe) [inline src/AAVE/Modal/BorrowModal.jsx:387]:
```suggestion
                        left: <PurpleTexture>Health Factor</PurpleTexture>,
```

File: src/AAVE/Modal/RepayModal.jsx [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+            deadline,
+            sig.v,
+            sig.r,
+            sig.s
+          ).then((tx) => {
+            tx.wait().then((res) => {
+              const { status } = res;
+              if (status === 1) {
+                onActionSuccess({
+                  msg: `You repaied ${Big(amount)
```
Comment (think-in-universe) [inline src/AAVE/Modal/RepayModal.jsx:273]:
```suggestion
                  msg: `You repaid ${Big(amount)
```

File: src/AAVE/Modal/RepayModal.jsx [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+          {
+            value: amount,
+          }
+        )
+        .then((tx) => {
+          tx.wait().then((res) => {
+            const { status } = res;
+            if (status === 1) {
+              onActionSuccess({
+                msg: `You repaied ${Big(amount)
```
Comment (think-in-universe) [inline src/AAVE/Modal/RepayModal.jsx:325]:
```suggestion
                msg: `You repaid ${Big(amount)
```

**Review by think-in-universe — CHANGES_REQUESTED** [review]

File: src/AAVE/Modal/BorrowModal.jsx [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+ * @returns {BigNumber}
+ */
+function borrowAllowance(vwETHAddress, userAddress) {
+  const vToken = new ethers.Contract(
+    vwETHAddress,
+    config.vwethABI.body,
+    Ethers.provider().getSigner()
+  );
+
+  // TODO replace address
```
[the remark this answers is not in the document]
Comment (SteveYuOWO) [reply src/AAVE/Modal/BorrowModal.jsx]: Forgot deleted. Already replaced. The address is a constant `0xabc...` value before.

File: src/AAVE.jsx [hunk 6]
```diff
@@ -333,12 +537,98 @@ function updateData() {
           yourSupplies,
         });
 
-        if (JSON.stringify(prevYourSupplies) === JSON.stringify(yourSupplies)) {
+        if (
+          JSON.stringify(prevYourSupplies) === JSON.stringify(yourSupplies) &&
+          JSON.stringify(prevYourBorrows) === JSON.stringify(yourBorrows)
```
Comment (think-in-universe) [inline src/AAVE.jsx:542]: Is LINE 542 necessary for updating supplies data?

**Review by think-in-universe — CHANGES_REQUESTED** [review]

**Review by think-in-universe — APPROVED** [review]

**Review by think-in-universe — APPROVED** [review]

**Pull request merged**